src/game/common/skirmishbattlehonors.cpp

```diff
@@ -27,3 +39,617 @@ int SkirmishBattleHonors::Get_Endurance_Medal(Utf8String name, int level) const
+        image = g_theMappedImageCollection->Find_Image_By_Name("HonorBlitz10");
+        Insert_Battle_Honor(listbox, image, 0, BLITZ_10, &row, &column, 0, 0);
+    } else {
+        image = g_theMappedImageCollection->Find_Image_By_Name("HonorBlitz10");
+        Insert_Battle_Honor(listbox, image, 1, BLITZ_10, &row, &column, 0, 0);
+    }
+
+    int bestWinStreak = BattleHonors->Get_Best_Win_Streak();
+    Utf16String str;
+    str.Format(U_CHAR("%10d"), bestWinStreak);
+    if (bestWinStreak >= 1000) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("HonorStreak_1000");
+        Insert_Battle_Honor(listbox, image, 1, STREAK, &row, &column, 0, bestWinStreak);
+    } else if (bestWinStreak >= 500) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("HonorStreak_500");
+        Insert_Battle_Honor(listbox, image, 1, STREAK, &row, &column, 0, bestWinStreak);
+    } else if (bestWinStreak >= 100) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("HonorStreak_100");
+        Insert_Battle_Honor(listbox, image, 1, STREAK, &row, &column, 0, bestWinStreak);
+    } else if (bestWinStreak >= 25) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("HonorStreak_G");
+        Insert_Battle_Honor(listbox, image, 1, STREAK, &row, &column, 0, bestWinStreak);
+    } else if (bestWinStreak >= 10) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("HonorStreak_S");
+        Insert_Battle_Honor(listbox, image, 1, STREAK, &row, &column, 0, bestWinStreak);
+    } else if (bestWinStreak >= 3) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("HonorStreak_B");
+        Insert_Battle_Honor(listbox, image, 1, STREAK, &row, &column, 0, bestWinStreak);
+    } else {
+        image = g_theMappedImageCollection->Find_Image_By_Name("HonorStreak_B");
+        Insert_Battle_Honor(listbox, image, 0, STREAK, &row, &column, 0, bestWinStreak);
+    }
+
+    int wins = BattleHonors->Get_Wins();
+    str.Format(L"%10d", wins);
+    if (wins >= 10000) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("Domination_10000");
+        Insert_Battle_Honor(listbox, image, 1, DOMINATION, &row, &column, 0, wins);
+    } else if (wins >= 1000) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("Domination_1000");
+        Insert_Battle_Honor(listbox, image, 1, DOMINATION, &row, &column, 0, wins);
+    } else if (wins >= 500) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("Domination_500");
+        Insert_Battle_Honor(listbox, image, 1, DOMINATION, &row, &column, 0, wins);
+    } else if (wins >= 100) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("Domination_100");
+        Insert_Battle_Honor(listbox, image, 1, DOMINATION, &row, &column, 0, wins);
+    } else {
+        image = g_theMappedImageCollection->Find_Image_By_Name("Domination_100");
+        Insert_Battle_Honor(listbox, image, 0, DOMINATION, &row, &column, 0, wins);
```

Comment:
When changing `char` to `bool`, please change third argument to true/false accordingly in all calls.